pkg/k8s/apis/cilium.io/v2/cegp_types.go

```diff
@@ -63,7 +63,17 @@ type CiliumEgressGatewayPolicySpec struct {
 	ExcludedCIDRs []CIDR `json:"excludedCIDRs"`
 
 	// EgressGateway is the gateway node responsible for SNATing traffic.
+	// In case multiple nodes are a match for the given set of labels, the first node
+	// in lexical ordering based on their name will be selected.
 	EgressGateway *EgressGateway `json:"egressGateway"`
+
+	// Optional list of gateway nodes responsible for SNATing traffic.
+	// The first entry must contain the gateway node from the EgressGateway field.
```

Comment:
This seems confusing to me, why not just have the EgressGateways field take precedence over the old singular one.  Such that if EgressGateways is not nil then we simply ignore EgressGateway.

EgressGateways just seems like a generalization of EgressGateways anyway so in my mind I would expect have EgressGateways of length one to be equivalent to using EgressGateway (assuming they're the same egress-gateway).